### PR TITLE
Added data availability checks

### PR DESCRIPTION
**Pull request by vbuterin**
Replacement for #1083

### CODE REVIEW

File: specs/core/2_data-availability-proofs.md [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+        coordinates = [(index, proof.axis_index) for index in proof.indices]
+    elif proof.checking_extension:
+        # Case 2: proof is getting indices along a row, we are checking against the extension root
+        generalized_indices.append(get_generalized_index(DataAvailabilityProof, 'extension', proof.axis_index - CHUNKS_PER_ROW))
+        coordinates = [(proof.axis_index, index) for index in proof.indices]
+    else:
+        # Case 3: proof is getting indices along a row, we are checking against the row root
+        generalized_indices.append(get_generalized_index(DataAvailabilityProof, 'rows', proof.axis_index))
+        coordinates = [(proof.axis_index, index) for index in proof.indices]
+    # Each individual element can be come from the row/extension trees, or from the column tree
```
Comment (vbuterin) [inline specs/core/2_data-availability-proofs.md]:
```suggestion
    # Each individual element can be come from the row/extension trees, or from the column tree
    assert proof.indices == sorted(set(proof.indices))
```

**Comment by dankrad** [comment]
Should we agree to call it "data availability checks" for the future?
I believe the name "data availability proof" has already caused a lot of confusion, because it implies someone is proving something to someone else but that's actually not the best way to think about it.

@vbuterin let me know if you agree with this and I would be happy to change it

**Comment by vbuterin** [comment]
> @vbuterin let me know if you agree with this and I would be happy to change it

Sounds reasonable to me! Agree that data availability checks is better than proofs.

File: specs/core/2_data-availability-proofs.md [hunk 2]
```diff
@@ -0,0 +1,280 @@
+# Ethereum 2.0 Phase 2 -- Data Availability Proofs
```
Comment (vbuterin) [inline specs/core/2_data-availability-proofs.md]:
```suggestion
# Ethereum 2.0 Phase 2 -- Data Availability Checks
```

File: specs/core/2_data-availability-proofs.md [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+
+**Notice**: This document is a work-in-progress for researchers and implementers.
+
+## Table of contents
+
+<!-- TOC -->
+
+- [Ethereum 2.0 Phase 2 -- Data Availability Proofs](#ethereum-20-phase-2----data-availability-proofs)
```
Comment (vbuterin) [inline specs/core/2_data-availability-proofs.md]:
```suggestion
- [Ethereum 2.0 Phase 2 -- Data Availability Checks](#ethereum-20-phase-2----data-availability-checks)
```

File: specs/core/2_data-availability-proofs.md [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+        - [`get_data_sqiare`](#get-data-square)
+        - [`extend_data_square`](#extend-data-square)
+        - [`mk_data_root`](#mk-data-root)
+        - [`process_data_extension_slashing`](#process-data-extension-slashing)
+
+<!-- /TOC -->
+
+## Introduction
+
+This document describes the expected formula for calculating data availability proofs and the beacon chain changes (namely, slashing conditions) needed to enforce them.
```
Comment (vbuterin) [inline specs/core/2_data-availability-proofs.md]:
```suggestion
This document describes the expected formula for calculating the Merkle root used for data availability checks and the beacon chain changes (namely, slashing conditions) needed to enforce them.
```

**Comment by hwwhww** [comment]
replaced by #2146

**Pull request closed without merging**